csvs/dmmbitcoin-cashflow.json

```
@@ -0,0 +1,303 @@
+            },
+            "comment": {
+              "type": "join",
+              "values": [
+                "口座種別: ",
+                { "type": "cell", "column": "口座種別" },
+                " / 入出金の種別: ",
+                { "type": "cell", "column": "入出金の種別" },
+                " / 数量: ",
+                { "type": "cell", "column": "数量" },
+                { "type": "cell", "column": "" },
+                " / 備考: ",
+                { "type": "cell", "column": "備考" }
+              ]
+            },
+            "transfers": []
+          }
+        ]
+      },
+      {
+        "description": "CB",
+        "conditions": [
+          {
+            "value": {
+              "type": "cell",
+              "column": "入出金の種別"
+            },
+            "pattern": "キャッシュバック"
+          },
+          {
+            "value": {
+              "type": "cell",
+              "column": ""
+            },
+            "pattern": "JPY"
+          }
+        ],
+        "generators": [
+          {
+            "type": "cash-back",
+            "source": {
+              "type": "join",
+              "values": [
+                { "type": "service" },
+                "_cashflow_",
+                { "type": "cell", "column": "日時" }
+              ]
+            },
+            "timestamp": {
+              "type": "cell",
+              "column": "日時"
+            },
+            "comment": {
+              "type": "join",
+              "values": [
+                "入出金の種別: ",
+                { "type": "cell", "column": "入出金の種別" },
+                " / 数量: ",
+                { "type": "cell", "column": "数量" },
+                { "type": "cell", "column": "" },
+                " / 備考: ",
+                { "type": "cell", "column": "備考" }
+              ]
+            },
+            "transfers": [
+              {
+                "direction": "in",
+                "amount": {
+                  "type": "text",
+                  "value": {
+                    "type": "cell",
+                    "column": "数量"
+                  }
+                },
+                "asset": {
+                  "type": "symbol",
+                  "value": { "type": "cell", "column": "" }
+                }
+              }
+            ]
+          }
+        ]
+      },
+      {
+        "description": "出金手数料",
+        "conditions": [
+          {
+            "value": {
+              "type": "cell",
+              "column": "入出金の種別"
+            },
+            "pattern": {
+              "type": "reg-exp",
+              "pattern": "出金手数料$"
+            }
+          },
+          {
+            "value": {
+              "type": "cell",
+              "column": "口座種別"
+            },
+            "pattern": "ウォレット"
+          },
+          {
+            "value": {
+              "type": "cell",
+              "column": "数量"
+            },
+            "pattern": {
+              "type": "nonpositive"
+            }
+          },
```

```suggestion
```

下と確認内容被ってるかも

---

これゼロ比較するためです。正でなく負でない実数はゼロ。
そのコメント書いたのですがデバッグしている間に消してしまったようです。明日対応します

---

これゼロ比較するためです。正でなく負でない実数はゼロ。
そのコメント書いたのですがデバッグしている間に消してしまったようです。明日対応します

---

```suggestion
          {
            "value": {
              "type": "cell",
              "column": "数量"
            },
            "pattern": "\"0\""
          },
```

なるほど、見間違えてました、上 nonpositive, 下 nonnegative か
ゼロ比較なら文字列直接代入して検証できるのでこっちの書き方の方が良いかも

---

そのアプローチやったんですが、bignumber変換後の処理になるからかマッチしませんでした。ダブルクオートが重要？